tls-2pc-aio/src/prf/mod.rs

```diff
@@ -0,0 +1,181 @@
+        let a1 = hmac_sha256(&ms, &seed);
+        let a2 = hmac_sha256(&ms, &a1);
+        let mut a1_seed = [0u8; 109];
+        a1_seed[..32].copy_from_slice(&a1);
+        a1_seed[32..].copy_from_slice(&seed);
+        let mut a2_seed = [0u8; 109];
+        a2_seed[..32].copy_from_slice(&a2);
+        a2_seed[32..].copy_from_slice(&seed);
+        let p1 = hmac_sha256(&ms, &a1_seed);
+        let p2 = hmac_sha256(&ms, &a2_seed);
+        let mut ek = [0u8; 40];
+        ek[..32].copy_from_slice(&p1);
+        ek[32..].copy_from_slice(&p2[..8]);
+
+        let mut cwk = [0u8; 16];
+        cwk.copy_from_slice(&ek[..16]);
+        let mut swk = [0u8; 16];
+        swk.copy_from_slice(&ek[16..32]);
+        let mut civ = [0u8; 4];
+        civ.copy_from_slice(&ek[32..36]);
+        let mut siv = [0u8; 4];
+        siv.copy_from_slice(&ek[36..]);
+        (cwk, swk, civ, siv)
+    }
+
+    #[ignore = "expensive"]
+    #[tokio::test]
+    async fn test_prf() {
+        let (leader_channel, follower_channel) = DuplexChannel::<PRFMessage>::new();
+        let (gc_leader, gc_follower) = mock_dualex_pair();
+        let leader = PRFLeader::new(Box::new(leader_channel), gc_leader);
+        let follower = PRFFollower::new(Box::new(follower_channel), gc_follower);
+
+        let client_random: [u8; 32] = thread_rng().gen();
+        let server_random: [u8; 32] = thread_rng().gen();
+
+        let (leader_share, follower_share) = get_shares().await;
+
+        let pms = leader_share + follower_share;
+
+        let (task_leader, task_follower) = tokio::join!(
+            tokio::task::spawn_blocking(move || {
+                futures::executor::block_on(leader.compute_session_keys(
+                    client_random,
+                    server_random,
+                    leader_share,
+                ))
+            }),
+            tokio::task::spawn_blocking(move || {
+                futures::executor::block_on(follower.compute_session_keys(follower_share))
+            })
+        );
+
+        let (leader_keys, _leader) = task_leader.unwrap().unwrap();
+        let (follower_keys, _follower) = task_follower.unwrap().unwrap();
+
+        let cwk = leader_keys
+            .cwk()
+            .iter()
+            .zip(follower_keys.cwk())
+            .map(|(a, b)| a ^ b)
+            .collect::<Vec<u8>>();
+        let swk = leader_keys
+            .swk()
+            .iter()
+            .zip(follower_keys.swk())
+            .map(|(a, b)| a ^ b)
+            .collect::<Vec<u8>>();
+        let civ = leader_keys
+            .civ()
+            .iter()
+            .zip(follower_keys.civ())
+            .map(|(a, b)| a ^ b)
+            .collect::<Vec<u8>>();
+        let siv = leader_keys
+            .siv()
+            .iter()
+            .zip(follower_keys.siv())
+            .map(|(a, b)| a ^ b)
+            .collect::<Vec<u8>>();
+
+        let (expected_swk, expected_cwk, expected_siv, expected_civ) =
```

Comment:
There must be an error somewhere because the correct return order is cwk, swk, civ, siv